apps/marketplace/components/Opportunities/OpportunitiesFilters.js

```diff
@@ -0,0 +1,465 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import AUaccordion from '@gov.au/accordion/lib/js/react.js'
+import { AUcheckbox } from '@gov.au/control-input/lib/js/react.js'
+import AUheading from '@gov.au/headings/lib/js/react.js'
+import styles from './Opportunities.scss'
+
+export class OpportunitiesFilters extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      accordionOpen: props.accordionOpen,
+      statusFilters: {
+        live: false,
+        closed: false
+      },
+      openToFilters: {
+        selected: false,
+        one: false,
+        all: false
+      },
+      typeFilters: {
+        innovation: false,
+        outcomes: false,
+        training: false,
+        specialists: false
+      },
+      activeStatusFilters: {
+        live: false,
+        closed: false
+      },
+      activeOpenToFilters: {
+        selected: false,
+        one: false,
+        all: false
+      },
+      activeTypeFilters: {
+        innovation: false,
+        outcomes: false,
+        training: false,
+        specialists: false
+      }
+    }
+    this.handleCheckboxChange = this.handleCheckboxChange.bind(this)
+    this.handleTypeFilterClick = this.handleTypeFilterClick.bind(this)
+    this.handleFilterCancelClick = this.handleFilterCancelClick.bind(this)
+    this.handleFilterApplyClick = this.handleFilterApplyClick.bind(this)
+  }
+
+  setActiveFilterState() {
+    this.setState(curState => {
+      const newState = { ...curState }
```

Comment:
could this not work like `getFilterValues` ?

Reply:
by referencing `this.state`? I believe when setting state you want to avoid references to `this.state`.